src/test/java/org/tugraz/sysds/test/functions/builtin/BuiltinL2SVMTest.java

```diff
@@ -0,0 +1,125 @@
+    private final static int colsX = 500;
+    private final static int colsY = 1;
+    private final static double spSparse = 0.01;
+    private final static double spDense = 0.7;
+    private final static int max_iter = 10;
+
+    @Override
+    public void setUp() {
+        TestUtils.clearAssertionInformation();
+        addTestConfiguration(TEST_NAME,new TestConfiguration(TEST_CLASS_DIR, TEST_NAME,new String[]{"C"}));
+    }
+
+    @Test
+    public void testL2SVMDense() {
+        runL2SVMTest(false, 0, eps, 1.0, max_iter, LopProperties.ExecType.CP);
+    }
+    @Test
+    public void testL2SVMSparse() {
+        runL2SVMTest(true, 0, eps, 1.0, max_iter, LopProperties.ExecType.CP);
+    }
+//
+    @Test
+    public void testL2SVMIntercept() {
+        runL2SVMTest(true,1, eps, 1.0, max_iter, LopProperties.ExecType.SPARK);
+    }
+//
+    @Test
+    public void testL2SVMDenseIntercept() {
+         runL2SVMTest(false,1, 1, 1.0, max_iter, LopProperties.ExecType.CP);
+    }
+
+    @Test
+    public void testL2SVMSparseLambda2() {
+        runL2SVMTest(true,1, 1, 2.0, max_iter, LopProperties.ExecType.CP);
+    }
+
+    @Test
+    public void testL2SVMSparseLambda100CP() {
+        runL2SVMTest(true,1, 1, 100, max_iter, LopProperties.ExecType.CP);
+    }
+    @Test
+    public void testL2SVMSparseLambda100Spark() {
+        runL2SVMTest(true,1, 1, 100, max_iter, LopProperties.ExecType.SPARK);
+    }
+
+    @Test
+    public void testL2SVMIteration() {
+        runL2SVMTest(true,1, 1, 2.0, 100, LopProperties.ExecType.CP);
+    }
+
+    private void runL2SVMTest(boolean sparse, int intercept, double eps,
+                               double lambda, int run, LopProperties.ExecType instType)
+    {
+        Types.ExecMode platformOld = setExecMode(instType);
+
+        boolean oldFlag = OptimizerUtils.ALLOW_ALGEBRAIC_SIMPLIFICATION;
+        boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
+
+        try
+        {
+            loadTestConfiguration(getTestConfiguration(TEST_NAME));
+
+            double sparsity = sparse ? spSparse : spDense;
+
+            String HOME = SCRIPT_DIR + TEST_DIR;
+
+            fullDMLScriptName = HOME + TEST_NAME + ".dml";
+            programArgs = new String[]{ "-explain", "-stats",
+                    "-nvargs", "X=" + input("X"), "Y=" + input("Y"), "model=" + output("model"),
+                    "inc=" + intercept,"eps=" + eps, "lam=" + lambda, "max=" + run};
+
+
+            fullRScriptName = HOME + TEST_NAME + ".R";
+            rCmd = getRCmd(inputDir(), Integer.toString(intercept), Double.toString(eps),
+                    Double.toString(lambda), Integer.toString(run), expectedDir());
+
+            //generate actual datasets
+            double[][] X = getRandomMatrix(rows, colsX, 0, 100, sparsity, 10);
+            double[][] Y = getRandomMatrix(rows, 1, -1, 1, 1, -1);
+            for(int i=0; i<rows; i++)
+                Y[i][0] = (Y[i][0] > 0) ? 1 : -1;
```

Comment:
Instead of doing this you can use `Y = TestUtils.round(rand(0,1))` which the script should be able to convert to -1/1.

Reply:
I have incorporated all the suggested changes.
looking forward for your valuable feedback.